src/fsharp/ExtensionTyping.fs

```diff
@@ -330,7 +331,7 @@ module internal ExtensionTyping =
         member __.Namespace = x.Namespace
         member __.FullName = x.FullName
         member __.IsArray = x.IsArray
-        member __.Assembly = x.Assembly |> ProvidedAssembly.Create
+        member __.Assembly: ProvidedAssembly = x.Assembly |> ProvidedAssembly.Create
```

Comment:
Is it necessary for the Assembly property to be typed ProvidedAssembly or is Assembly enough?

Reply:
> Is it necessary for the Assembly property to be typed ProvidedAssembly or is Assembly enough?

if we do not explicitly specify the Assembly type, then we cannot refer to its FullName property in the ProvidedTypeComparer code above, since the types are inferred later :(